Dockerfile

```
@@ -1,86 +1,381 @@
+ARG POWERSHELL_VERSION=7.2.8
+ARG POWERSHELL_DOWNLOAD_URL=https://github.com/PowerShell/PowerShell/releases/download/v$POWERSHELL_VERSION/powershell-$POWERSHELL_VERSION-linux-x64.tar.gz
+ARG POWERSHELL_DOWNLOAD_SHA=28FF2653667AC63B508F0B98433E48F64E6BC1EC59F8C1D252BA89EB5A7441A2
+
+RUN set -ex \
+    && curl -SL $POWERSHELL_DOWNLOAD_URL --output powershell.tar.gz \
+    && echo "$POWERSHELL_DOWNLOAD_SHA powershell.tar.gz" | sha256sum -c - \
+    && mkdir -p /opt/microsoft/powershell/$POWERSHELL_VERSION \
+    && tar zxf powershell.tar.gz -C /opt/microsoft/powershell/$POWERSHELL_VERSION \
+    && rm powershell.tar.gz \
+    && ln -s /opt/microsoft/powershell/$POWERSHELL_VERSION/pwsh /usr/bin/pwsh
+#****************     END .NET-CORE     *******************************************************
+
+
+#****************      NODEJS     ****************************************************
+
+ENV NODE_18_VERSION="18.15.0"
+
+RUN  n $NODE_18_VERSION && npm install --save-dev -g -f grunt && npm install --save-dev -g -f grunt-cli && npm install --save-dev -g -f webpack \
+     && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+     && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+     && apt-get update && apt-get install -y -qq --no-install-recommends yarn \
+     && yarn --version \
+     && cd / && rm -rf $N_SRC_DIR && rm -rf /tmp/*
+
+#****************      END NODEJS     ****************************************************
+
+#**************** RUBY *********************************************************
+
+ENV RUBY_31_VERSION="3.1.3"
+
+RUN rbenv install $RUBY_31_VERSION && rm -rf /tmp/* \
+    && rbenv global $RUBY_31_VERSION && ruby -v
+
+#**************** END RUBY *****************************************************
+
+#**************** PYTHON *****************************************************
+# inspired by: https://github.com/aws/aws-codebuild-docker-images/blob/9282872af78aeb1b5df3010ed3872c40f3d0f056/al2/x86_64/standard/2.0/Dockerfile
+ENV PYTHON_38_VERSION="3.8.10"
+
+ARG PYTHON_PIP_VERSION=21.1.2
+ENV PYYAML_VERSION=5.4.1
+
+COPY tools/runtime_configs/python/$PYTHON_38_VERSION /root/.pyenv/plugins/python-build/share/python-build/$PYTHON_38_VERSION
+RUN   env PYTHON_CONFIGURE_OPTS="--enable-shared" pyenv install $PYTHON_38_VERSION && rm -rf /tmp/*
+RUN   pyenv global  $PYTHON_38_VERSION
+RUN set -ex \
+    && pip3 install --no-cache-dir --upgrade --force-reinstall "pip==$PYTHON_PIP_VERSION" \
+    && pip3 install --no-cache-dir --upgrade "PyYAML==$PYYAML_VERSION" \
+    && pip3 install --no-cache-dir --upgrade setuptools wheel aws-sam-cli boto3 pipenv virtualenv
+
+#**************** END PYTHON *****************************************************
+
+#****************      PHP     ****************************************************
+ENV PHP_81_VERSION="8.1.13"
```

nit: should we remove the quotes here?
```suggestion
ENV PHP_81_VERSION=8.1.13
```